### PR TITLE
chore: add homepages for each learn, build and user

### PR DESCRIPTION
**Pull request by samricotta**
## Description

Update to include home pages for learn, build and user - all versioned to make user navigation easier. Additionally I went ahead and added all of the dropdown's from the tabs up the top so all of the tabs at the top are uniform

## Related Issue
#95, #96, #97

## Changes Summary

- Added in homepage for each learn, build, user
- Updated the sidebars to manual to ensure that we can use the homepage as I needed to implement a temp solution (also this gives us a bit more control over the sidebars) 
- Most of the changes(449) are just because I updated from 0.50 to v0.50 across the docs to ensure everything is uniform. Additionally, it became essential when making the homepage additions

## Testing

Yes, with npm start. Went through all of the pages and branches to ensure there are no broken links. Additionally ran the script to check if there are issues with the script. 

## Screenshots 

Screenshots to show the changes 

Screenshot 2023-12-11 at 10 16 17
Screenshot 2023-12-11 at 10 16 07
Screenshot 2023-12-11 at 12 06 00
Screenshot 2023-12-11 at 12 06 10

### CODE REVIEW

File: docs/learn/learn.md [hunk 1]
```diff
@@ -0,0 +1 @@
+package learn
```
Comment (samricotta) [inline docs/learn/learn.md:1]: Havent figured out a way of adding a `docsId` to the pages/learn yet. Tried inserting some react but this is a bit of a hacky workaround for now to ensure the versioning works.

File: docusaurus.config.js [hunk 2]
```diff
@@ -4,7 +4,8 @@
 const lightCodeTheme = require("prism-react-renderer/themes/github");
 const darkCodeTheme = require("prism-react-renderer/themes/dracula");
 
-const lastVersion = "0.50";
+
+const lastVersion = "v0.50";
```
Comment (samricotta) [inline docusaurus.config.js:9]: updated all of these so they are consistent

File: sidebars.js [hunk 3]
```diff
@@ -14,9 +14,152 @@
 /** @type {import('@docusaurus/plugin-content-docs').SidebarsConfig} */
 const sidebars = {
   // By default, Docusaurus generates a sidebar from the docs folder structure
-  learnSidebar: [{ type: "autogenerated", dirName: "learn" }],
-  buildSidebar: [{ type: "autogenerated", dirName: "build" }],
-  userSidebar: [{ type: "autogenerated", dirName: "user" }],
+  "learnSidebar": [
```
Comment (samricotta) [inline sidebars.js:17]: all these had to be changed to ensure that the learn/learn page above (where i commented about the versioning) is not possible to get to

**Comment by julienrbrt** [comment]
Nice!

Can we get the homepage `/` category, to point to those category homepages?
It feels a bit weird to have a "Home" option in the dropdowns. I haven't tried, but it looks like, we could make the dropdown clickable: https://docusaurus.io/docs/api/themes/configuration#navbar-dropdown, which could then link to the homepag instead of having that "Home" link.
Additionally, for the components, could we make them more reusable and reduce the code duplication?

**Comment by samricotta** [comment]
- yep! ill ensure the homepage categories link to those
- Yeah, the first thing I tried was to make the whole dropdown clickable but unfortunately it didnt work, which is why I used the home option (i also think its a little weird) but the only work around. Ill have another go
- Yep, good shout as a lot of them are duplicated

File: docusaurus.config.js [hunk 4]
```diff
@@ -98,33 +102,115 @@ const config = {
...
+                type: "doc",
+                label: "Build Apps",
+                docId: "build/building-apps/app-go",
+              },
               {
                 type: "doc",
                 label: "Build a Module",
                 docId: "build/building-modules/intro",
               },
+              {
```
Comment (tac0turtle) [inline docusaurus.config.js:152]: whats the thought behind doing a dropdown for everything?

**Comment by tac0turtle** [comment]
learn dropdown is odd since once you click on one field all fields are highlighted, maybe we dont need to have every item in the sidebar in the drop down? 


What was the rationale behind a different style page instead of a readme in the folder as a simple explainer whats in the section with links? The flow feels weird to me

**Comment by samricotta** [comment]
So using a readme, the best it can look is:

Screenshot 2023-12-12 at 10 57 01

or 

Screenshot 2023-12-12 at 11 06 15

The list works for the modules but these for the main pages (learn, build, user), I dont think are aesthetically nice, nor nice for the user. But thats personal opinion, hence why I took it to the next level. But happy to change to whatever the consensus is on this. 

Also I think when we used the wording "landing page", what I built is what comes to mind. I would call the other options contents pages

- Dropdowns I dont mind so much, but I based this off the near docs. They have all of them also listed, only having a few I feel isnt uniform but also gives the illusion we dont have as many docs. 
Screenshot 2023-12-12 at 11 11 20

- Yeah the highlighting is something thats a bit strange, its because of the preexisting style package. Trying to figure out why its doing that but need to timebox

Happy to just remove the dropsdowns tho after that, give the above a read and let me know what I should do. Maybe @julienrbrt you have some thoughts on this too.

**Comment by tac0turtle** [comment]
i think the readme is sufficient. Its a bit more of a simple look, which is fine because it reduces maintenance in the long run

**Comment by samricotta** [comment]
So both examples are readme's. Should example 1 (list) or example (2) index be implemented? 

Moving forward lets try ensure that we are a bit clearer regarding expectations. As I mentioned above landing page is what was in asked in issue #57 but what is expected now is more of a list, which could have been created in a fraction of the time.

**Comment by tac0turtle** [comment]
the feedback from noam was a place to see what is all under the sections. the same thing was shared with you that was with me. Creating a simple design is all thats needed. Lets first sketch things out next time together before starting to work to avoid doing work then changing

**Comment by samricotta** [comment]
Yep totally agree and let's do that moving forward. Going back to the question, both examples I gave are readme's. Should example 1 (Module summary example) or example 2 (index example) be implemented?

**Comment by tac0turtle** [comment]
is the second automatic ot?

**Pull request closed without merging**